File: 9-palindrome-number/9-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
             if(x < 0){
            return false;
        }
        
        string s = to_string(x);
        
        for(int i=0; i<s.length(); i++){
            
            if(i >= s.length()-1-i){
                break;
            }
            if(s[i] != s[s.length()-1-i]){
                return false;
            }
        }
        return true;
        
    }
};
        
 

File: 9-palindrome-number/9-palindrome-number.java
class Solution {
public boolean isPalindrome(int x) {
int rev = 0;
int rem;
int temp = x;

    while(temp>0){
        rem = temp%10;
        rev = rev*10 + rem;
        temp = temp/10;
    }
    
    System.out.println(rev);
    if(x==rev){
        return true;
    }
    else{
        return false;
    }
    
}
}